api/envoy/api/v2/discovery.proto

```diff
@@ -55,6 +55,14 @@ message DiscoveryRequest {
   // internal exception related to the failure. It is only intended for consumption during manual
   // debugging, the string provided is not guaranteed to be stable across Envoy versions.
   google.rpc.Status error_detail = 6;
+
+  // [#not-implemented-hide:]
+  // If true, unsubscribes to all resources. This is intended to be used in LDS/CDS for clients using the resource_names
+  // field that want to unsubscribe from the last resource. For RDS/EDS, this can be done by sending a request with an
+  // empty resource_names field, but for LDS/CDS, an empty resource_names field triggers wildcard behavior, so we need an
+  // alternative mechanism to unsubscribe from the last resource. This may be useful if (e.g.) a client is using ADS but is
+  // being reconfigured from a full LDS/RDS/CDS/EDS flow to a simple EDS-only flow.
```

Comment:
This seems a fairly unlikely scenario.

Reply:
I agree, it's definitely not going to be a common case.  But there doesn't seem to be another way to handle it in the API.

We can live without adding this if you'd prefer.

Reply:
I'm trying to understand if this is motivated by a realistic situation that you have or it's some corner case that is contrived as an example? Ideally we have examples that come from the former.

Reply:
This is something that we ran into while implementing the gRPC xDS client code.  No one has actually run into this yet in practice.  If you'd prefer to defer this until someone does run into it, I'm fine with that.

Reply:
I'd drop the sentence for now; other examples I can think of are contrived, as there is no client that implements them.

Reply:
Maybe let's close this one out for now then; my original thinking was that it made sense to add this as gRPC would need it, but based on conversations it seems this is less the case. I think this can be added later if needed.